lib/livebook_web/live/session_live.ex

```diff
@@ -1112,6 +1120,17 @@ defmodule LivebookWeb.SessionLive do
     {:noreply, socket}
   end
 
+  def handle_event("secrets", %{"ref" => ref, "current_secret" => current_secret}, socket) do
+    {:noreply,
+     push_patch(socket,
+       to:
+         Routes.session_path(socket, :secrets, socket.assigns.session.id,
+           ref: ref,
```

Comment:
This is an internal aspect of the flow, so instead of passing it in the URL, let's just assign it as `select_secret_ref` right away.